packages/client/src/index.ts

```diff
@@ -12,61 +11,60 @@ export const extractBearerToken = (authorization: string): string => {
     typeof authorization !== 'string' ||
     (!authorization.startsWith('Bearer ') && !authorization.startsWith('bearer '))
   ) {
-    return null;
+    throw new Error('Fail to extract bearer token');
   }
 
   const token = authorization.slice(7);
   return token;
 };
 
 export const ensureBasicOptions = (options?: ConfigParameters): ConfigParameters => {
-  const { clientId, discoveryUrl, redirectUris } = options || {};
-  if (typeof discoveryUrl !== 'string' || !discoveryUrl.startsWith('http')) {
-    throw new Error('Invalid discoveryUrl');
+  const { clientId, logtoUrl } = options || {};
+  if (typeof logtoUrl !== 'string' || !logtoUrl.startsWith('http')) {
+    throw new Error('Invalid logtoUrl');
   }
 
   if (typeof clientId !== 'string' || clientId.length === 0) {
     throw new Error('Need clientId');
   }
 
-  if (redirectUris) {
-    if (!Array.isArray(redirectUris)) {
-      throw new TypeError('RedirectUris should be an array of strings');
-    }
-
-    for (const uri of redirectUris) {
-      if (typeof uri !== 'string') {
-        throw new TypeError('RedirectUris should be an array of strings');
-      }
-    }
-  }
-
   return {
     clientId,
-    discoveryUrl,
-    redirectUris,
+    logtoUrl,
   };
 };
 
+export const ensureTrailingSlash = (url: string): string => {
+  if (url.endsWith('/')) {
+    return url;
+  }
+
+  return url + '/';
+};
+
 export class LogtoClient {
   public oidcReady = false;
-  public issuer: Issuer<Client>;
-  private client: Client;
+  public issuer: Issuer<Client> | null = null;
+  private client: Client | null = null;
   private readonly clientId: string;
-  private readonly redirectUris: string[];
   constructor(config: ConfigParameters, onOidcReady?: () => void) {
-    const { discoveryUrl, clientId, redirectUris } = ensureBasicOptions(config);
+    const { logtoUrl, clientId } = ensureBasicOptions(config);
     this.clientId = clientId;
-    this.redirectUris = redirectUris;
 
-    void this.initIssuer(discoveryUrl, onOidcReady);
+    void this.initIssuer(
+      `${ensureTrailingSlash(logtoUrl)}oidc/.well-known/openid-configuration`,
+      onOidcReady
+    );
   }
 
   public getClient(): Client {
+    if (!this.issuer) {
+      throw new Error('should init first');
```

Comment:
shall we use codes (constants) for errors first (like extract the constants [in a file](https://github.com/logto-io/logto/blob/master/packages/phrases/src/locales/en.ts)), for easier i18n in the future?

no hurry for i18n. we can add an issue first.

Reply:
sure, I'll create a seperate issue in linear.